src/rgen/Microsoft.Macios.Generator/Emitters/EnumEmitter.cs

```diff
@@ -177,4 +187,84 @@ public bool TryEmit (in BindingContext bindingContext, [NotNullWhen (false)] out
 
 		return true;
 	}
+
+	bool TryEmitErrorCode (in BindingTypeData<SmartEnum> bindingTypeData, in BindingContext bindingContext, [NotNullWhen (false)] out ImmutableArray<Diagnostic>? diagnostics)
+	{
+		diagnostics = null;
+		if (bindingContext.Changes.BindingType != BindingType.SmartEnum) {
+			diagnostics = [Diagnostic.Create (
+					Diagnostics
+						.RBI0000, // An unexpected error occurred while processing '{0}'. Please fill a bug report at https://github.com/xamarin/xamarin-macios/issues/new.
```

Comment:
Could we get a better error message? We know what went wrong. If this just shouldn't happen because otherwise something else went wrong, it's still useful to have something to search our code for ("An unexpected error occurred while processing '{0}': <something unique to this particular code site").